### DISCUSSION TITLE
How long does it usually take for api to refresh the cache with new user data? After SignUp?

### DISCUSSION THREAD

**Discussion by Mayank-Tripathi32** [open-sauced-craftwork/browser-extension#50] [Q&A] [answered]
Hey, I am trying to add SignUp feature, After authenticating with github and completing onboarding,

![image](https://user-images.githubusercontent.com/70465598/228713808-2d4f5fee-847c-4b9d-8fdd-9e8fbab3b7c8.png)
![image](https://user-images.githubusercontent.com/70465598/228713897-6903a166-c0b0-41a7-b4db-7a3f8407e21c.png)

But even then https://api.opensauced.pizza/v1/users/Mizety returns 404, So How long should i add the cooldown for new users before they access the ext, or how should i deal with this?

**Comment by 0-vortex** [comment, marked as answer]
There is no caching right now, users get added to the database on logging in, it happens in the api here: [src/auth/auth.controller.ts#L44](https://github.com/open-sauced/api.opensauced.pizza/blob/cae5dbdc28cd409decfcdad9effb13b860f35be0/src/auth/auth.controller.ts#L44)

**Comment by Mayank-Tripathi32** [comment]
@0-vortex Apprently, until you sign up again at https://hot.opensauced.pizza/# manually , api wont work 🤔 even tho im using same supabase keys to sign up from extension.

**Reply by 0-vortex** [reply]
> @0-vortex Apprently, until you sign up again at https://hot.opensauced.pizza/# manually , api wont work 🤔 even tho im using same supabase keys to sign up from extension.

Set up your own supabase account like instructed in the docs, so you can understand the authz model that is provided. The code linked above is reading the authz and then storing that data in our user tables. Since you don't have access to our database, you cannot do that programatically, that is part of why this assignment is requiring innovative and sometimes counter-intuitive thinking! 🍕
